aws-distro-opentelemetry-node-autoinstrumentation/src/patches/extended-instrumentations/aws-sdk-instrumentation-extended.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk';
+import { context as otelContext, defaultTextMapSetter } from '@opentelemetry/api';
+import { AWSXRayPropagator } from '@opentelemetry/propagator-aws-xray';
+import type { Command as AwsV3Command } from '@aws-sdk/types';
+
+const awsXrayPropagator = new AWSXRayPropagator();
+const V3_CLIENT_CONFIG_KEY = Symbol('opentelemetry.instrumentation.aws-sdk.client.config');
+type V3PluginCommand = AwsV3Command<any, any, any, any, any> & {
+  [V3_CLIENT_CONFIG_KEY]?: any;
+};
+
+// This class extends the upstream AwsInstrumentation to override its patching mechanism of the `send` method.
+// The overriden method will additionally update the AWS SDK middleware stack to inject the `X-Amzn-Trace-Id` HTTP header.
+//
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+export class AwsSdkInstrumentationExtended extends AwsInstrumentation {
+  // Override the upstream private _getV3SmithyClientSendPatch method to add middleware to inject X-Ray Trace Context into HTTP Headers
+  // https://github.com/open-telemetry/opentelemetry-js-contrib/blob/instrumentation-aws-sdk-v0.48.0/plugins/node/opentelemetry-instrumentation-aws-sdk/src/aws-sdk.ts#L373-L384
+  override _getV3SmithyClientSendPatch(original: (...args: unknown[]) => Promise<any>) {
+    return function send(this: any, command: V3PluginCommand, ...args: unknown[]): Promise<any> {
+      this.middlewareStack?.add(
```

Comment:
Q: is Middleware Stack existed on all AWS supported AWS JS SDK versions?  it learned it was introduced by AWS JS SDK V3 on 2020. 

https://aws.amazon.com/blogs/developer/modular-aws-sdk-for-javascript-release-candidate/

Reply:
This PR will add support for v3.1.0
- `@aws-sdk/smithy-client ['^3.1.0']` (for aws-sdk packages before from `3.1.0` to `v3.363.0`)
- `@smithy/smithy-client ['>=1.0.3']` (for aws-sdk packages `v3.363.0+`)

>is Middleware Stack existed on all AWS supported AWS JS SDK versions
>learned it was introduced by AWS JS SDK V3 on 2020.

Yes, that is also near the beginning of AWS SDK JS V3 span. See version history:
- https://www.npmjs.com/package/@aws-sdk/client-s3?activeTab=versions
- https://www.npmjs.com/package/@aws-sdk/client-lambda?activeTab=versions
```
v3.1.0		54	4 years ago
v3.0.0		834	4 years ago
v1.0.0-rc.10	66	4 years ago
```

Reply:
So there is no support for `v3.0.0`, but that's fine since [OTel doesn't support `v3.0.0` either](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/instrumentation-aws-sdk-v0.48.0/plugins/node/opentelemetry-instrumentation-aws-sdk/src/aws-sdk.ts#L85-L164)